content/articles/creating-movies-and-animations-using-matlab/index.md

```diff
@@ -0,0 +1,243 @@
+
+### Prerequisites
+To follow along with this tutorial, you will need:
+- [MATLAB](https://www.mathworks.com/products/get-matlab.html?s_tid=gn_getml) installed.
+- Proper understanding of [MATLAB](https://www.section.io/engineering-education/getting-started-with-matlab/) basics.
+
+Animation in Matlab follows a workflow like that of creating a flipbook. The steps involved in creating an animation in Matlab are as follows:
+1. Run a simulation or generate data.
+2. Draw/render the scenario at some time `t_k`.
+3. Take a snapshot of the scenario.
+4. Advance time `t_k` to `t_(k+1)`.
+
+We will then, Go back and repeat steps 2 to 4 and keep going through building one frame or one page in the flipbook one at a time and saving it to the large flipbook. This means that the last step is:
+
+5. Saving the movie.
+
+### Matlab functions that can be important at each of the steps
+#### 1. Run a simulation or generate a data
+Maybe you have a fancy flight simulator that will run a scenario, kick out all this data, and save it to a file. So all you need here is to load the data. 
+
+It means that You use the `load` function here. Also, if you are familiar with Simulink, you know that you can run a Simulink model from a Matlab script to generate data using the `sim` function.
+
+#### 2. Drawing/ rendering the scenario at some point t_x
+This deals with plotting or drawing one single frame of the animation. It means that you will use the plot functions such as `plot`, `plot3`, `surf`. 
+
+A couple of things that can be helpful include the` hold on` function for complicated scenarios or animations. Also, it helps to draw many figures on the same plot.
+
+Because of jamming the `plot` function inside the `for` loop, you have to periodically wipe the slate clean after every time. In this, we use the `clf` function.
+
+#### 3. Take a snapshot of the scenario
+Once you have drawn one page of the flipbook, we want to grab that frame and save it into the large flipbook. Matlab has the function `get frame` for doing this.
+
+#### 4. Advance time t*k to t*(k+1)
+Like we said before, If we put the process inside a `for` loop or a `while` loop, the step is automatically handled. If you have data that you are simulating that is very temporally timely spaced, you might have tons of data. 
+
+You may not want to plot every single point of your data since that will be a very dense movie. So we will implement the logic of skipping some data and using the `continue` function for this.
+
+#### 5. Saving the movie
+To save the movie, we will use `VideoWriter` and `WriteVideo` functions.
+
+### Example
+We want to animate the trajectory of a point. Let us say we run some simulations or have some equations that generate the trajectory at some point in the space. 
+
+The position vector describing where the point is located at any given `x`, `y`,`z` location given time `t` is:
+
+$$
```

Comment:
You need to add a code snippet here.

Reply:
This is just a mathematical equation and not a code and it is using special markdown character. If we add the snippets it wont work.